### PR TITLE
Adding setting to control MCNP / ENDF library

### PR DESCRIPTION
**Pull request by john-science**
## What is the change?

Adding setting `mcnpLibraryBaseName` to control ENDF/B library name for MCNP.

## Why is the change being made?

To make it easier to control the elements we use to model.

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [x] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify any new/changed code.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/developer/tooling.html#add-release-notes) have been updated if necessary.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

File: armi/physics/neutronics/settings.py [hunk 1]
```diff
@@ -172,6 +173,16 @@ def defineSettings():
...
             enforcedOptions=True,
         ),
+        setting.Setting(
+            CONF_MCNP_LIB_BASE,
+            default="ENDF/B-VII.1",
+            description=(
+                "Library name for MCNP cross sections. "
+                "ENDF/B-VII.1 is the default library. "
+            ),
+            label="Default base library name",
```
Comment (onufer) [inline armi/physics/neutronics/settings.py:183]: `ENDF version to use for MCNP Analysis`

File: armi/physics/neutronics/settings.py [hunk 2]
```diff
@@ -43,6 +43,7 @@
 CONF_INNERS_ = "inners"
 CONF_LOADING_FILE = "loadingFile"
 CONF_NEUTRONICS_KERNEL = "neutronicsKernel"
+CONF_MCNP_LIB_BASE = "mcnpLibraryBaseName"
```
Comment (onufer) [inline armi/physics/neutronics/settings.py:46]: `mcnpLibraryVersion`

File: armi/physics/neutronics/settings.py (hunk 1 above)
Comment (john-science) [reply armi/physics/neutronics/settings.py]: done

File: armi/physics/neutronics/settings.py (hunk 2 above)
Comment (john-science) [reply armi/physics/neutronics/settings.py]: done

**Review by onufer — APPROVED** [review]

File: armi/reactor/blueprints/isotopicOptions.py [hunk 3]
```diff
@@ -536,22 +535,20 @@ def eleExpandInfoBasedOnCodeENDF(cs):
 
     if "MCNP" in cs[CONF_NEUTRONICS_KERNEL]:
         expansionStrings.update(mcnpExpansions)
-        if int(cs["mcnpLibrary"]) == 50:
```
Comment (john-science) [inline armi/reactor/blueprints/isotopicOptions.py:539]: This needs to be another option for the new setting.

**Pull request merged**